renderer-macos/lib/stubs/BriskScrollView.c

```diff
@@ -9,6 +13,52 @@
   return NO;
 }
 
+- (void)setOnScroll:(value)callbackOption {
+  if (self._callback) {
+    value prevCallback = self._callback;
+    caml_remove_global_root(&prevCallback);
+  }
+  if (Is_block(callbackOption)) {
+    self.contentView.postsBoundsChangedNotifications = YES;
+    [[NSNotificationCenter defaultCenter] 
+      addObserver:self 
+      selector:@selector(boundsDidChange) 
+      name:NSViewBoundsDidChangeNotification 
+      object:self.contentView]; 
+    value callback = Field(callbackOption, 0);
+    caml_register_global_root(&callback);
+    self._callback = callback;
+  } else {
+    self._callback = 0;
+    [[NSNotificationCenter defaultCenter] 
+      removeObserver:self 
+      name:NSViewBoundsDidChangeNotification
+      object:self.contentView]; 
+    self.contentView.postsBoundsChangedNotifications = NO;
+  }
+}
+
+- (void)boundsDidChange {
+  if (self._callback) {
+    NSRect visibleRect = [self.contentView documentVisibleRect];
+    NSSize contentSize = self.contentSize;
+    dispatch_async(dispatch_get_main_queue(), ^{
+    value args[] = {
+        caml_copy_double(visibleRect.origin.x),
```

Comment:
We should investigate if this plays well with OCaml GC/allocator.

Reply:
I'm on it!

Reply:
https://github.com/janestreet/ecaml/blob/ae67c7ffa8a9e5610b3b42f3385b3cd513595217/ecaml_value/src/ecaml_stubs.c#L641
We don't need to.